### DISCUSSION TITLE
Use clindata::rawplus_rdsl in Map_Raw functions

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#145] [General]
Quick discussion regarding how to use the new `clindata::rawplus_rdsl` data set in Map_Raw. We can create issues once we agree on approach here. 

## `AE_Map_Raw`
  - `dfExposure` becomes `dfRDSL`  
  -  add `strExposureCol` with default of `"TimeOnTreatment"`

## `PD_Map_Raw`
  - `dfTOS` becomes `dfRDSL` 
  - add `strExposureCol` with default of `"TimeOnStudy"`

## `IE_Map_Raw` 
Probably no changes needed right now, but I suppose we could add dfRDSL so that we get accurate participant counts at sites. That information is not used in the pipeline, but is used in the draft visualization.  

## `Consent_Map_Raw`
- `dfIxrsrand` becomes `dfRDSL` for `RandDate`

## `Disp_Map_Raw`

Will use rdsl a lot running this, but don't think any updates are needed, other than maybe updating the examples

@gwu05 does this make sense? Anything I'm forgetting?

**Comment by kodesiba** [comment]
For AE_Map_Raw we need to either add back the Exposure column in the clindata function for numeric exposure or remove it as a requirement in AE_Map_raw as the two currently do not work with eachother

**Comment by gwu05** [comment]
Agree with the list of needed updates. I think if dfRDSL is available that should be the dataset we use as the source of SubjectID and SiteID (this would need to be updated), and as you mentioned for specific functions, the required columns can be specified from dfRDSL (e.g. strRandCol = "RandDate" as default, etc.). This way if a user wants to analyze a different subset of patients or create their own custom dfRDSL, they can create a data frame with SubjectID, SiteID, and "Needed subject info" (e.g. TImeOnStudy), then simply identify the column variable they derived.

So general idea is: each function requires raw datasets specific to the outcome of interest + subject level dataset (dfRDSL) with required subject info.

For Disp Map Raw, it should directly be derivable from dfRDSL as the only required input, with the caveat that we may need to tweak a bit what is dervied/required by Disp_Map_Raw and what is derived/provided as part of dfRDSL testing on some example cases.
